app/Http/Controllers/Table/FdbTablesController.php

```diff
@@ -0,0 +1,270 @@
+            }
+            if ($this->getMacCount($fdb_entry->port) == 1) {
+                // only one mac on this port, likely the endpoint
+                $item['interface'] .= ' <i class="fa fa-star fa-lg" style="color:green" aria-hidden="true" title="' . __('This indicates the most likely endpoint switchport') . '"></i>';
+            }
+        }
+
+        return $item;
+    }
+
+    /**
+     * @param string $ip
+     * @return Builder
+     */
+    protected function findMacs($ip)
+    {
+        $port_id = \Request::get('port_id');
+        $device_id = \Request::get('device_id');
+
+        return Ipv4Mac::where('ipv4_address', 'like', "%$ip%")
+            ->when($device_id, function ($query) use ($device_id) {
+                $query->where('device_id', $device_id);
+            })
+            ->when($port_id, function ($query) use ($port_id) {
+                $query->where('port_id', $port_id);
+            })
+            ->pluck('mac_address');
+    }
+
+    /**
+     * @param string $vlan
+     * @return Builder
+     */
+    protected function findVlans($vlan)
+    {
+        $port_id = \Request::get('port_id');
+        $device_id = \Request::get('device_id');
+
+        return Vlan::where('vlan_vlan', $vlan)
+            ->when($device_id, function ($query) use ($device_id) {
+                $query->where('device_id', $device_id);
+            })
+            ->when($port_id, function ($query) use ($port_id) {
+                $query->whereIn('device_id', function ($query) use ($port_id) {
+                    $query->select('device_id')->from('ports')->where('port_id', $port_id);
+                });
+            })
+            ->pluck('vlan_id');
+    }
+
+    /**
+     * @param string $ifAlias
+     * @return Builder
+     */
+    protected function findPorts($ifAlias)
+    {
+        $port_id = \Request::get('port_id');
+        $device_id = \Request::get('device_id');
+
+        return Port::where('ifAlias', 'like', "%$ifAlias%")
+            ->when($device_id, function ($query) use ($device_id) {
+                $query->where('device_id', $device_id);
+            })
+            ->when($port_id, function ($query) use ($port_id) {
+                $query->where('port_id', $port_id);
+            })
+            ->pluck('port_id');
+    }
+
+    /**
+     * @param string $mac_address
+     * @return \Illuminate\Support\Collection
+     */
+    protected function findIps($mac_address)
+    {
+        if (!isset($this->ipCache[$mac_address])) {
+            $ips = Ipv4Mac::where('ipv4_mac.mac_address', $mac_address)->pluck('ipv4_address');
+
+            $dns = 'N/A';
+            // don't try too many dns queries, this is the slowest part
+            foreach ($ips->take(2) as $ip) {
+                $hostname = gethostbyaddr($ip);
+                if (!IP::isValid($hostname)) {
```

Comment:
Better to cache all responses, valid or not, won't change per run anyway.

Reply:
good point

Reply:
@Jellyfrog I looked at the code and we cache the results for each mac every time.  So there shouldn't be any duplicated requests.

Reply:
Yeah I misread, sorry.
Wouldn't it make more sense to cache per IP instead of mac? Multiple mac-addresses could have the same IP (different networks).

Reply:
Also maybe making the DNS resolving a config option?